### PR TITLE
LSIF: Add page for job list over all repositories

### PR DESCRIPTION
**Pull request by efritz**
This provides a simple UI of LSIF jobs in the site-admin page. This UI is part of [RFC 46](https://docs.google.com/document/d/1v1Sojj4B07ZTX3bI3pu6uVCRGkYhujkQXmIi6A3Nbc4).

Screen Shot 2019-11-10 at 10 47 31 AM
Screen Shot 2019-11-10 at 10 47 38 AM

### CODE REVIEW

File: web/src/site-admin/SiteAdminLsifJobsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    private subscriptions = new Subscription()
+
+    public componentDidMount(): void {
+        eventLogger.logViewEvent('SiteAdminLsifJobs')
+
+        this.subscriptions.add(
+            this.updates
+                // Do not set statsOrError as null here to indicate loading
+                // so that the stats don't disappear during navigation, which
+                // causes some werid jitter.
```
Comment (attfarhan) [inline web/src/site-admin/SiteAdminLsifJobsPage.tsx]:
```suggestion
                // causes some weird jitter.
```

File: web/src/site-admin/SiteAdminLsifJobsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        },
+    ]
+
+    public state: State = {
+        statsOrError: null,
+        hideInternal: true,
+    }
+
+    private toggles = new Subject<void>()
+    private updates = new Subject<void>()
```
Comment (felixfbecker) [inline web/src/site-admin/SiteAdminLsifJobsPage.tsx]: It would be good to add docblocks to these two explaining what _events_ they represent

**Comment by efritz** [comment]
@felixfbecker  see https://github.com/sourcegraph/sourcegraph/pull/6567 for changes.

File: web/src/site-admin/SiteAdminLsifJobsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                    location={this.props.location}
+                    listClassName="list-group list-group-flush mt-3"
+                    cursorPaging={true}
+                />
+            </div>
+        )
+    }
+
+    private onToggle = (hideInternal: boolean): void => {
+        const toggles = this.toggles
```
[the remark this answers is not in the document]
Comment (efritz) [reply web/src/site-admin/SiteAdminLsifJobsPage.tsx]: I had an issue with this/bind. It seems fine.

File: web/src/site-admin/SiteAdminLsifJobsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+} from '../components/FilteredConnection'
+
+interface ToggleComponentProps {
+    hideInternal: boolean
+    onToggle: (enabled: boolean) => void
+}
+
+const ToggleComponent: FunctionComponent<ToggleComponentProps> = ({ hideInternal, onToggle }) => (
+    <div className="lsif-jobs-filter-toggle">
+        <label className="radio-buttons__item lsif-jobs-filter-toggle-label" title="Hide internal jobs">
```
[the remark this answers is not in the document]
Comment (efritz) [reply web/src/site-admin/SiteAdminLsifJobsPage.tsx]: data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxAQDw8PEBAQDw8PDQ4ODw0PEBAQDw4OFREWFhURFRUYHSggGBolGxUVIj0hJikrLy4uFx8/ODMsNygtLisBCgoKDg0OGBAQFysdHx0vLTcrLSsrKystLS8rLS0tKy0rLSstLSs3LS0tKzc3LS0tLSsrLS0rKystLS0tLy0tLf/AABEIAMcA/QMBIgACEQEDEQH/xAAcAAACAwEBAQEAAAAAAAAAAAABAgADBAUGBwj/xABHEAABAwICBQgFCwEGBwEAAAABAAIDBBESIQUxUWGRBhMiQVJxodEUgZKi4QcjMkJTYmNyscHwFTRDVIKTwjNkg6OlstIW/8QAGgEBAQEBAQEBAAAAAAAAAAAAAAECAwQFBv/EACoRAAICAQQBAgUFAQAAAAAAAAABAhEDBBIhMUFhoSIycYGRE1HR4fAj/9oADAMBAAIRAxEAPwDys9PaQ2ORvY/dOrwXnqaiLp8DZWseZm2cSWua5z7WbtPduXtNGaEqZ4ISInAhgaS8Fn0cgelbqsulQ8k5Y5WPLowBI1zg54vYEeK+bp1lg5VBvx0fVzyxZscd00mizTWhwyN8kdxJgLS8HpuGGxN+p29fOaHRzXueCWvPSdHfFi1W6eHVYr7RUU7nMtZrsupzV4x+heZc88yYy5xu7DYuF7jPrTI5YE5JNWefBtzNRkzLooumpGMqWh74S6Noe0Ehow8cutVVVI2MWZk0ZhvZ3D1rok4Mm5EdLb6u5U6QeMJyy+HxXznkc5blwm+vB9LDjWOaj+Gb4NKQYW4pY2mwuC9oIy6wn/qtP9vF/qNXzfS0dpbjCMTGuzBOYu3/AGrMB95nAr9BFXFM+LlW2co/s2fUv6tTfbxf6jVP6xTfbxe21fLh+ZnApv8AMzgtbUY3H046Zpvt4vbCU6dpR/fx+0F8z/zN9lQ/nb7Km1Dcz6Y/lDSgE8811hezcydwXgwb52td7jbZcqMYLautN1D8y0lRG7KDJhyIOWwXQdUjLJ2rZvWmU9L1hZpcOV8Q2YetKIL6SOy7gFPSfuu8Evzf4h9bkfm+zJxeqQcVB7DvBMKg9g8VWBH2JOMnmmsz7KT/ALvmgLBM7sePwRMz+wPa+CrAZ9g//ueaNm/4d3CTzSxQHSv7I4/BVumf2RxVtm/4c+y7zQwj/De4gKTK/Y3ilMr/ALi0Yf8Alx7DVLf8uPYYhTPzru0wfzvU549tngtHS+wHBil3/ZD3Eohm5/8AFb4Ic9+KPDyWvHJ2Bxapil7LfaCAx87+J4fBDnfxDw+C2Ey7G8ULy7G8VQerpuVNVOHmSS1nN6LOiA0g/uE080gjkkYXOOAgkuOXrXm9HvIc4Drb4gg+a7VPPkBe4OtvVdfMz6bPlyOUU5I+pg1Gnx4ds5KL9R6HTFTJCHAAuuW3BIOXWtlJp+uhGFwM4JvaQdG3ZzT00nUALbLLYyKR2ocBdYjj1kJcQaX1/s4PJo31kT/30LoGw1YJY11PN9aF30JDsa46l53S8jmlzXDC5pILTkb+S61TTSXuXWtqdY5EetNpSkNVBjd/aYG9J4Gc0I15bQu0dDkyLc8e1/an78P2OmLX6fDNXk3cetr25XueIq6EzBrm4btu031nO/7qmHRsjDfCx1jexsR4rv0tIMJBzzBVrqBpB7l9COGWOKjNU0eHNmhlyPJjdpnn9JzxAtdzbBLgsY2gYQe0V5+eS51jM9WSaVuGR7T1OcOBWaRudlzSpGW7LXAi1+tDErJpMQaLZt69qpQp6aN3RHq/RFxyHeqqc9Bv5W/orHah3rbIWxXLR3bQnsf4QqofojPwVmW3w+KIWGx/hCmE7vaCFhtPD4o2G08PihA4TtHtBHCdo9oJbDaeA80bDaeA80AcJ2j2lMB2t9pCzdp4DzUs3aeA80BCw7RxQ5s7W8UbDaeA81LDaeA80AMG9vFTBvbxRsNp4DzUsNp4DzQAwbxxPkhg3jx8k1htPAeamEbTwVAuDePHyQLd48U9t54IEbz4IBMO8eKmHf8Aqmw7z4KYd58EBQzJzTsPgu7QxFzmtAuTlYC54Lo6G5EySNEtS70aEDEb2Erh3HJo3ngvQU2mKOnHN0UYdYG85zvbL6RzP6L0aWco7uOzy6zHGSi26oxw8nKqQlwvRiKQYJJXMkbPH9YujGbd2a7ckVLEImSyPkfcyNLLxh5A+79UA6rrzWnKSescHOqJRFzdzCDhjLyRbIa8r676lkgDnuMJe8PpiHQvvcgOFi07RuXKXCc+6OsNvEY8WesrtMMa04KVkgAuDI5tiRqFzdUaN0/JI5rX0sMQcOohxsRqtZfP+UMErXFw51uL+7a68JdtaNY7l2eRlTM5wZKx942l2N7SMLR1XOexWGWM1aJKEovwdyTS1I5745KJoIeWl0JDSbEi+VlbW8nWk/MSWfhDvR5rNfY6rOWDkvT8/WOeR0GudK64ytiyHH9FzncoJJNJvxYhDI58TGWIdzYyDxlqyvfeu+pqM6+hy0knPHf1PA8pKJ8FXKyRhYcZdhcLGx6+7eqGUxkIDWk5C5Fyvo1XKyoLdHaWFpCXtotJAAOBvYXO/Lcd2Sy0/Jz0TDHVvZGcbmNLZA3nm36Lm36jcLxSrs9as8izQbzkCLFtwXZJJNATDrZ7S+k0mhYGuIzdlduJwPR2LLJpWhic5jom426musC47jqssb4G9rPGRRlrQ06wAD3hF2r1rVpCVr5ZHsbha5xIaMwNyyu1esLp4MDwEYRlt27VZcbPEpKcnDr6z171Zc7fFEGDENg8UcQ2DifNQF233gmu7b7wQC4hsHE+aOIbBxPmjd233h5o9Lb7w80AuMbBxPmjiGwcT5punt94eanT2+8PNALibsHE+amJuwcT5p+n/CPNTp/whAJibsHE+al27PEpjj3+CnS2HgEAMtniVMv4Sj0th4KXOz3VQDL+EoED+Eo33e6pfd7qAWw/hKlh/CVL7vBS+7wQHU07p2erPTcRH1RNyYO/aVXoOQglu/IbevzWV7P0T0WT8tn6LenleVX5OOrh/wAn6HrJK4sha3AbYrufe9h1Eri6Srw2ojmYGhpa4OwXuRfLF1cNq6FY35uIRyMBewc5ckhpO/qO5aqSnhZG1vNsdhGTjYm+9azYnK1F8M5Y8qilfZw6/SjSW6sWtod+p3L0M82CnMjspJ2gNb1hlsyqp9E01/S6luBgA6JcRzpGpobsQoA6vqWXADNdhqbEDqW9LhUeX0uzOpytpKPcuv5N9Peh0Y+UWbPUEBhIvhByBt3XK8pR1hjLntu+V/0pHWLnbtw3BdL5S690odDBkIcLG4SG5hwxWPqt6l845qs/E9Ug81y1ablb8no0rShtj0uD3PK7R8tfBDJG2z4mu+aJsSDbIOPXkn0XTyaWofQp7x19Gb080gIMkepzHHryy7wCqNBacIgbHMxwcGhtznmBruuTQaVnh0lFUAPLGyhjrA4TG7ouP7+pcb4s614Pb6X5OT0lBG+Kf010Ra2dvRZzbXfRLeuwOWe1eU//ADVXWnoU7SQLA8/9EH1WC+oRuaypdE4fMVTCx2yzx5lcnknTzUgqKd5JdDVyta52ZfH0Sx1/ykKZYxj8SXBYNvhnnNH/ACZ6RwgO9HZYfXmJ/wDVpXTZ8k9YR0p6Zt8+jzrv9oXt26Tf15rUzTL7WsLWsizY3xyHCXoeCZ8lMzR0quLWdUTj+6B+TF3+Mj/0Xf8A0voDtJYhYi3chzjD2vBbtPolPyfPX/Jo8aqyI/8ASf5ql3yczdVTCe8PavpTYmHrd4eStbQxHW5/u+Sy45H0W4+T5RL8n1UNUsDu57v3asNVyOrI8+bLxtjMbvDFfwX2xmjYfxDwCJoIh9S/5iSkcebzRHOB+d5IsJLXYmuBsWuZYg7CCUMI7R9n4r0fyhFv9RmAaMmxDK4+oNnqXm8Q7I4u81tqnRE7DhHa934qYR2vAoY29kcXeaHON2e8VCjYPvDgVMP3h4pcbdnvFTG3YfaQDYd48fJTCdo8fJKHN38R5I3G/iPJUDYTtHE+SGE7uKFxtPgpcbT4ICFp3cULH+FHFvPAIYt/gqD1+jooY43CcQ4w7EwOljOdhrLb7Fn0TRs54uvDObOOBjgSCesNe0NK9UyGL6sLDvfd/gncyS1hhjbsYxrAvJDTxjKUr7+/4Orm2kqOdHo9jgHSQMgiw2LpjGxwIN7gN1+GSom0tQwuOCMTOys5rGhrSBlZb6mgbI0NeOcsScThk3LUFnGgKf7Jq9MP0o/MmcprI/laOb/WhPKOca7CMo3YWF7HEnV6rBepnqYqCEPcDz01h9FvOlu122wsq9HaBp4CJ3RgObnG3PXttdYtLUUdTJzkoLjaws94AbsABW28aalT/JhLJTVr8HznldpKNxexoOJ78VshgbivmB1rzbHd6+pS8k6MknmAScyS59yeKV3Jel+waO5zh+6xPKmywxbUfMw87TxUMru07iV9HPJim1CLg5/mtFDyKp5HtGB1gQXHG6wAWN1m6OvpjlLTUkNIKqOSSR0DJGBgF8hYgm4sm5PcpW6SM8wi5kMe2INJxOcMAOJx2529Sw8qtDQ1kwLi8CNgiZhIAwj1bStPJTQzKNsjGOc4PeHnHa4NrdQ3KZX8LQgubO6CrWriHlFStLhJIYcLi3FPHJEwkOI6L3DC4XB1FdGmr4ZBeOaKQHUWSMcDwK8sUzszcxXxrMwrQxezGcZGuNa41kiWyLfl3r1xODNbVH9ajUH6iteTDPgvL+scNJ1QDXkB0Yu3Ufmmb15/05/Zl4fFey5WaDnlrqmRrQWuly6QBsGgbNy4zuTtR9n7zV5JyW5npinSOKa92yXgUp0ifxPZK67tA1H2R9pvmqzoWo+ydxb5rO5FpnL/AKiNrvW0+SH9Qbt90+S6LtEzj+7d+v7qs6Nm+yf7JKbkKZi9PZtHs/BT01m1vALU7R8n2T/Yd5Kp1G4a43eth8k3IUVemM2s8FPSmbW8UTTbWcW/BKYG9keCtgPpDdreKXnxtHH4oOhZ2W+CXmGbB4K2Sj7c2MOHzcuL7rnEFVSRvbrae8C44pIqENF3yMb3ZlLPpJkf/DmeSOogFq4cvs68eC+Bj36m+s6gtLXMj68b/dC57NPC1nYXeohO3SUB1st+Vy0qRGWTyOebn4BUkAa7BWGppz1yjuF/2RYKfqEzjvaf3CU3zZLSM5dfUL7+pKISTtOwLpNdGNUTj+ZwCc1LgLNEcfcMR4quK8sbvQzQaNcc32jYNZdrUqa9jGmOEE3+k4az6+pSSPGbvc5/5jlw1JHNaMh6gFU4r5SNN9mWGMk3d6mjUFtp9Z9SDYydeW4fuVYxtiuc38JpLk8/NJVvkbT00kUQaKyd7pY+dElqpzWx2uLDXc69S5+kZi+aOB2jKF05pG1EjKh0bXPeXOa6OFwacRGHXfrC7c2jpHnnYJRDURS1LA57OcjfHJJjcx7bjrwkEHIjeVh0noeqLImyRw6VYIMEkdQWQSCfETz0bw02uDa1/qjeucGjTMU4pGmmjg0XIaiphdN6MJDSOia02cHEkZ3vkuxoHRFHVRufzNZA+OV0UkMlXWNcyRtr2tJZwzGYXJm0JUNp6OOqpW6QZE2USCJ59Lp3F94+bkLm4mhuFpH3QvQch6epZHOJ2zMiM5NLFUSCWeOCwyc4E5X1AnJeqBxkdCHklR6y2c/mrKxw4GRHTHJuijppHsp2c40xlr3Ynua7nGi4LiTddyEKnlD/AGV++SnHGeML0xRxZ3Qg9MEHalswzwVe280p/Ef+pWcxrqzx3e87XuPiquZXzZPlntS4OdgSlq6RhSmFZstHNLUpbu8F0TAlMCWU5pYNg4BIWDYOC6RgVbqdZbBznNbs/VVFrez+q6L6ZUOplhyZpIxmNh1tCrNND2Bwb5LaadL6Op+oy7Ub20cXZv33P6q1sLBqAHqCRpTYlvcZHDQnFlViTApuFFt1MSruim4UPiQc+2tUumzwtGJ36d56kzIetxudn1R5puAzCXash2j19wVzIwO/b1lFqJKm4UFRBQI5cCji1+kJYJHc23nLvqzzPXK9tIyVjQdYzB4q7k1piWo/4jIiDG2VlRSvdLTuubGMkgFkgy6J/ZWTaNE0jyXvjMcz3NdGbO+coxFcHqIxXB2tCyaF0NUx1hnl9Ga30cxSOpw9rqt+JpbLKw5Bwsc7n6RVjRGepar41S1Xxr1YzlI1RLNykNqXvqqFvGshH7rVEFj5Uf2Zm+v0WP8AyEC9UTgz0LVH6lGqqpqGNIYXND3hxYwkYnBouSB1gLRg804ZnvKUtRaU1l8w95XhQwpylJUAuBAsT3UugKixKY1cglAoMSqdAtiBCm0tnOfAqzCumWpDGFl4y7jAAiAmATBqwUQBMAjZUOkLjhYL9Rd9Uevr7ggHlma0Z/wpWte/Xdjdn1j5K2GkAOI9J3aPV3DqWjClgrjYGiwFk102FTCpZaFxIgo4VA1ZbAbogohqYNSyHG09Wywsl5jDz8r6WOHHbDje4tud2Frj6kruVDWt5xzPmzoz09hvZznNdZ8VjqIxMHeSrtNaPbI9rpKYVUIDCYgW4mvaJG4sDrNeMMp67ggWBK89WaKp5G00EVSyIR1j3GnqcdPL6E+Rr5IGteLus5ot1W612hRmR6ek5TwPbSHDJerhnla0AHmxC28gfnkb5d4V2h+VUNSYxHDWBsubJX0sjYbWJuZPo2y13XF0VyfmZW1krgDTiGcUVnNNzUP5yUWBuLOFs+0qeRNFNB6LHJDpZj2MLXiSVjtHh2FwzZjJDc9muy9cDjI9JS8taRzo7NqeZlnbTx1no8gpXyl2ENEm92V7WVXKnlGSyRkFNLUR0ddRekTtcxrBNHURScwwE3e/IA2FhfWuDQaLq2OiZSUlbQTtqWukDaoP0U2LnbvIY55xAtv0Q0Zlb9IaJrYGzUUbqcQVmlmTw1bpDz8Jmna8tMGHpkG+eIZL0RbOTSOwOXQ9LcxrQ6mfohlfTvwuEj5nNLxCc7XLQbC2tpWbknNIXh00ofUf0mDnvmy+Z7sUxLnTamsu42b1nc1dmDkRTB8ZcXOZAdHGBmoxmjY9rLu+tfGbjJduWlZFA5kbQxrYi0AbA2wudZ9apng8vE8rQCkhiV2FeJo9SZWUpVhCUhZKV2QTkJShRULolAoCXQuoUqAa6CCl0KZgMkksob3nU0ZuPcEjXuf9DIfauGX+Ufv+q0Q04bnrcdbjm4+tcaotlDYHP+n0W9gHX+Y9fctTGACwFgOoJw1MGqCxQEbJgEwalFsSyOFOGpg1ShZVhRwK3CjhU2lsqwoOdZWOVJF1lxBa0oyRNeML2te3suAcOBVIu3quN2seauieDqN/2W4EZkZyco/q08cd9ZhBhPuEKxnJim21Q3CvrgOAlXQjWqNe7GcJGGn5MUo1tlfulqqqUcHyFdTRug6SB2OGmgifr5xkTA8n81rq2JbIl6oo4MvYqNIn5t28YeOSMlUxus59kZlc6oqS8i+QGpv796zKVEjGzNzVkjmq9yrIXjbPUkUFqQhXlqQtUKkUEJS1XlqUtUKUFqXCry1AtQFBalLVeQlIQFJCCsISkJYK2jJOAg1MFyo0EBMAgmCoIAmAUCYKAgamAUCKoJZAqEpCUSIByAahdMFqgMGhF0LTrA7+vigCmulIBZHbU5w9d/1VzXEfWPBvkqQ5EOW02jDRqbM7tHg3yTGQnW4ndc24LKHJg5dFN/uZ2ovuEWlUYlYwo3wVIvSlRQribEKQpylKAQhKUxSlAIUpTlKUAhSlMUpQCFKUxSlAVtTBI1OFgowTBIE4QDBMEoRugGUJQugUACUpRJSqoBRSoqgYFG6VRUg90bpEVQOCmBVYTLRB2lXtVLArmqNhFwUUChWCilIU5SFUCFKnKQoBSlKJSlAKUpRKUqAUpExSFCiNTBK1ELIHCYJAiCgHBTXSI3QDIFBRAQoIoKoARUQVA11EqKpBgUQUoTBUBCcJQFY0KgdiuakYFa1RgsUKN0CslEKQpykKEEKQpnJCqBSlKYpCgFKUolKVCikpCmKVAVsKa6KiyA3RugogGujdRRAS6N1FEAVFFFUCIKKLRCIhRRAFO0KKIBwFY0IKKguaE4UUWSliBUUQMQlISooqQrJSOKKiArJSEqKIBClJUUUApKW6iiFP/9k=

**Pull request merged**